app/stac_api/models.py

```diff
@@ -277,6 +280,31 @@ class Item(models.Model):
         null=False, blank=False, default=BBOX_CH, srid=4326, validators=[validate_geometry]
     )
 
+    class Meta:
+        unique_together = (('collection', 'name'),)
+        indexes = [
+            models.Index(fields=['name'], name='item_name_idx'),
+            # the following 3 indices are used e.g. in collection_temporal_extent
+            models.Index(fields=['properties_datetime'], name='item_datetime_idx'),
+            models.Index(fields=['properties_start_datetime'], name='item_start_datetime_idx'),
+            models.Index(fields=['properties_end_datetime'], name='item_end_datetime_idx'),
+            # created, updated, and title are "queriable" in the search endpoint
+            # see: views.py:322 and 323
+            models.Index(fields=['created'], name='item_created_idx'),
+            models.Index(fields=['updated'], name='item_updated_idx'),
+            models.Index(fields=['properties_title'], name='item_title_idx'),
+            # geometry is used in "filter_by_intersects", views.py:415 and filtering for bbox
+            models.Index(fields=['geometry'], name='item_geometry_idx'),
+            # combination of datetime and start_ and end_datetimes are used in
+            # managers.py:110 and following
+            models.Index(
+                fields=[
+                    'properties_datetime', 'properties_start_datetime', 'properties_end_datetime'
+                ],
+                name='item_dttme_start_end_dttm_idx'
+            ),
+        ]
+
```

Comment:
I would leave this block either after all class attribute definition or before them, but not in the middle. Because Collection and Asset already have the inner `Class Meta` at the top I would put it at the top.

Reply:
That was my intention when I shifted it from the bottom, where it currently is, to the top. Somehow pasted it at the wrong place though ;-)